openquake/hazardlib/geo/surface/simple_fault.py

```diff
@@ -256,16 +256,22 @@ def hypocentre_patch_index(cls, hypocentre, rupture_top_edge,
             contains the hypocentre.
         """
         totaln_patch = len(rupture_top_edge)
-        for index in range(1, totaln_patch):
-
+        indexlist = []
+        dist_list = []
+        for i, index in enumerate(range(1, totaln_patch)):
             p0, p1, p2, p3 = cls.get_fault_patch_vertices(
                 rupture_top_edge, upper_seismogenic_depth,
                 lower_seismogenic_depth, dip, index_patch=index)
 
             [normal, dist_to_plane] = get_plane_equation(p0, p1, p2,
                                                          hypocentre)
+            indexlist.append(index)
+            dist_list.append(dist_to_plane)
             if (numpy.allclose(dist_to_plane, 0., atol=25., rtol=0.)):
                 return index
+                break
+        index = indexlist[dist_list.index(min(dist_list))]
+        return index
```

Comment:
Are you trying to do something that could be done with numpy.argmin? http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.argmin.html